### DISCUSSION TITLE
LCX for local AI

### DISCUSSION THREAD

**Discussion by blablazzz** [tteck/Proxmox#1466] [Ideas]
Would be great to have LCX for selfhosted AI:

https://github.com/go-skynet/LocalAI

**Comment by mate1213** [comment]
I think there is a possibility to install LocalAi from source. (Thats a bit of a hassel, since I try to do it since 2 days and still not done yet)
These steps from the [original documentation](https://localai.io/basics/build/) of LocalAI:
```
git clone https://github.com/go-skynet/LocalAI
cd LocalAI
make build
```

I use Debian 12 base LXC image
I needed to install golang 1.21 from [here](https://go.dev/doc/install)
And the following packkges:
libprotobuf-dev
protobuf-compiler
libabsl-dev

Sadly I'm still stuck with protobuffer cannot be found by cmake...

Well, better luck than I had to figure out how can it run on LXC.

**Reply by AlexTryHarder** [reply]
Hi Mate,
I'm stuck in same place..
did you find a solution for this?

**Reply by mate1213** [reply]
Hi!
In the official repo i've got the info : should use ubuntu 22.04

I've tried but no luck.
https://github.com/mudler/LocalAI/discussions/1505#discussioncomment-7985534

**Comment by markyskus** [comment]
that would be greatm Im trying install it via docker with no luck :(

**Comment by manuelkamp** [comment]
docker works for me fine, but I want direct LXC (which I never got running)
